File: Lib/Telerik/Examples/QuickStart/Map/BingElevation/ExampleMapInputBehavior.vb
Imports Telerik.WinControls.UI
Imports Telerik.WinControls.UI.Map

Namespace Telerik.Examples.WinControls.Map.BingElevation
	Public Class ExampleMapInputBehavior
		Inherits MapInputBehavior
		Public Overrides Sub OnDoubleClick(ByVal e As EventArgs)
			Dim args As MouseEventArgs = TryCast(e, MouseEventArgs)

			Dim point As New PointL(args.X - Me.MapElement.PanOffset.Width, args.Y - Me.MapElement.PanOffset.Height)
			Dim location As PointG = MapTileSystemHelper.PixelXYToLatLong(point.X, point.Y, Me.MapElement.ZoomLevel)

            While location.Longitude > 180
                location.Longitude -= 360
            End While

			Dim pin As MapPin = TryCast(Me.MapElement.Layers("Pins").HitTest(location), MapPin)

			If pin IsNot Nothing Then
				Me.MapElement.Layers("Pins").Remove(pin)

				Return
			End If

			pin = New MapPin(location)
			Me.MapElement.Layers("Pins").Add(pin)
		End Sub
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/Map/BingElevation/ExampleMapInputBehavior.cs
using System;
using System.Windows.Forms;
using Telerik.WinControls.UI;
using Telerik.WinControls.UI.Map;

namespace Telerik.Examples.WinControls.Map.BingElevation
{
    public class ExampleMapInputBehavior : MapInputBehavior
    {
        public override void OnDoubleClick(EventArgs e)
        {
            MouseEventArgs args = e as MouseEventArgs;

            PointL point = new PointL(args.X - this.MapElement.PanOffset.Width, args.Y - this.MapElement.PanOffset.Height);
            PointG location = MapTileSystemHelper.PixelXYToLatLong(point.X, point.Y, this.MapElement.ZoomLevel);

            while (location.Longitude > 180)
            {
                location.Longitude -= 360;
            }

            MapPin pin = this.MapElement.Layers["Pins"].HitTest(location) as MapPin;

            if (pin != null)
            {
                this.MapElement.Layers["Pins"].Remove(pin);

                return;
            }

            pin = new MapPin(location);
            this.MapElement.Layers["Pins"].Add(pin);
        }
    }
}
